parser/parser.go

```diff
@@ -61,6 +63,16 @@ func (p *parser) next() lexer.Token {
 	return p.prev
 }
 
+func (p *parser) mark() {
```

Comment:
This is a bit shady, the parser is on the hot path so we should take care to make sure it remains fairly quick.